### PR TITLE
[ci] Improve instructions on how to replicate builds locally

### LINKED ISSUE #16242: CI "To replicate this build locally" not working (for me)
### Description

On a Alma 9.4 node (specifically cmslpc-el9-heavy01.fnal.gov), I used verbatim the instructions from the CI for an arbitrary PR build (see [here](https://github.com/root-project/root/actions/runs/10343508810/job/28658232103?pr=16202#step:9:8064) and a copy/paste in the reproducer.

However, I get 23 failures not seen on any of the CI runs (see list below and 
full log of the failures at: [RootCI-Container-LastTest.log](https://github.com/user-attachments/files/16615802/RootCI-Container-LastTest.log)

Looking in the log the problem seem to include:
 * Missing (py)Torch installation (PyMVA)
 * Missing ipython3 installation (Jupyter)
 * Failure to connect to a server (roottest-python-distrdf-backends-test_all )
 * Some more unclear failures
 * See  below for an extraction of the relevant output
 * see [RootCI-Container-LastTest.log](https://github.com/user-attachments/files/16615802/RootCI-Container-LastTest.log) for more details.

```
The following tests FAILED:
        385 - PyMVA-RandomForest-Classification (Failed)
        386 - PyMVA-RandomForest-Multiclass (Failed)
        387 - PyMVA-GTB-Classification (Failed)
        388 - PyMVA-GTB-Multiclass (Failed)
        389 - PyMVA-AdaBoost-Classification (Failed)
        390 - PyMVA-AdaBoost-Multiclass (Failed)
        391 - PyMVA-Torch-Classification (Failed)
        392 - PyMVA-Torch-Regression (Failed)
        393 - PyMVA-Torch-Multiclass (Failed)
        394 - gtest-tmva-pymva-test-TestRModelParserPyTorch (Failed)
        1068 - tutorial-tmva-TMVA_CNN_Classification (Failed)
        1069 - tutorial-tmva-TMVA_Higgs_Classification (Failed)
        1070 - tutorial-tmva-TMVA_RNN_Classification (Failed)
        1071 - tutorial-tmva-TMVA_SOFIE_PyTorch (Failed)
        1203 - tutorial-dataframe-df102_NanoAODDimuonAnalysis-py (Failed)
        1205 - tutorial-dataframe-df104_HiggsToTwoPhotons-py (Failed)
        1261 - tutorial-rcanvas-df104-py (Failed)
        1564 - roottest-python-JupyROOT-importROOT_notebook (Failed)
        1565 - roottest-python-JupyROOT-simpleCppMagic_notebook (Failed)
        1566 - roottest-python-JupyROOT-thread_local_notebook (Failed)
        1567 - roottest-python-JupyROOT-ROOT_kernel_notebook (Failed)
        1568 - roottest-python-JupyROOT-tpython_notebook (Failed)
        1664 - roottest-python-distrdf-backends-test_all (Failed)
```
The specific run I references only failed the `TMapFile` test (which sometimes fails, sometimes succeed).

The set of failure types is as follow:
```
Warning in <TMVA_CNN_Classification>: TMVA is not build with GPU or CPU multi-thread support. Cannot use TMVA Deep Learning for CNN
Running with nthreads  = 4
[37;41;1m<FATAL>                         : Can't import __main__[0m
***> abort program execution
```
```
Setup TMVA...
[37;41;1m<FATAL>                         : Can't import __main__[0m
***> abort program execution
```
```
Generate PyTorch model...
Traceback (most recent call last):
  File "/github/home/ROOT-CI/build/tmva/pymva/test/generatePyTorchModelClassification.py", line 1, in <module>
    import torch
ModuleNotFoundError: No module named 'torch'
```
```
RDataFrame::Run: event loop was interrupted
Traceback (most recent call last):
  File "/github/home/ROOT-CI/src/tutorials/dataframe/df102_NanoAODDimuonAnalysis.py", line 50, in <module>
    h.SetTitle("")
    ^^^^^^^^^^
cppyy.gbl.std.bad_alloc: TH1D& ROOT::RDF::RResultPtr<TH1D>::operator*() =>
    bad_alloc: std::bad_alloc
CMake Error at /github/home/ROOT-CI/build/RootTestDriver.cmake:232 (message):
  error code: 1
```
```
Traceback (most recent call last):
  File "/github/home/ROOT-CI/src/tutorials/dataframe/df104_HiggsToTwoPhotons.py", line 87, in <module>
    ROOT.RDF.RunGraphs([hists[s] for s in ["ggH", "VBF", "data"]])
cppyy.gbl.std.bad_alloc: unsigned int ROOT::RDF::RunGraphs(vector<ROOT::RDF::RResultHandle>) =>
    bad_alloc: std::bad_alloc
CMake Error at /github/home/ROOT-CI/build/RootTestDriver.cmake:232 (message):
  error code: 1
```
```
-- END TEST OUTPUT --
-- BEGIN TEST ERROR --
sh: 1: ipython3: not found
Traceback (most recent call last):
  File "/github/home/ROOT-CI/roottest/python/JupyROOT/nbdiff.py", line 197, in <module>
    retCode = canReproduceNotebook(nbFileName, kernelName, needsCompare)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
```
Exception occurred during processing of request from ('127.0.0.1', 53110)
FTraceback (most recent call last):
  File "/usr/lib/python3.12/socketserver.py", line 318, in _handle_request_noblock
    self.process_request(request, client_address)
```

### Reproducer

Grab the image:
```
  $ docker run --security-opt label=disable -it registry.cern.ch/root-ci/ubuntu2404:buildready
```
  Then:
```  
  (
  rm -rf /github/home/ROOT-CI
              mkdir -p /github/home/ROOT-CI
  )
  (
  curl --output /github/home/ROOT-CI/artifacts.tar.gz https://s3.cern.ch/swift/v1/ROOT-build-artifacts/ubuntu2404/master/RelWithDebInfo/3f796869dae3bde0bfafc4ef8051339eb5ca133c/2024-08-12.tar.gz
  )
  (
  cd /github/home/ROOT-CI && tar -xf /github/home/ROOT-CI/artifacts.tar.gz
  )
  (
  cd '/github/home/ROOT-CI/src'
                  git checkout master
                  git fetch
                  git reset --hard @{u}
  )
  (
  cd '/github/home/ROOT-CI/src'
        git fetch https://github.com/root-project/root 9fc748389d42ce698fe8654ac177f9cf9a542f5c
  )
  (
  cd '/github/home/ROOT-CI/src'
  
          git config user.email "rootci@root.cern"
          git config user.name 'ROOT Continous Integration'
  
          git fetch origin 7d45ab2957efacf7b6685f4ae126a9fed243494c:fix-15733
          git checkout fix-15733
          git rebase 531620f59e898e5ec809043135583d15b0d151a6
  )
  (
  git clone --branch master --single-branch https://github.com/root-project/roottest.git "/github/home/ROOT-CI/roottest"
  )
  (
  cd '/github/home/ROOT-CI/roottest'
  
          git config user.email "rootci@root.cern"
          git config user.name 'ROOT Continous Integration'
  
          git fetch  
          git checkout 
          git rebase master
  )
  (
  which cmake
          cmake --version
          which c++ || true
          c++ --version || true
          uname -a || true
          cat /etc/os-release || true
          sw_vers || true
          uptime || true
          df || true
  )
  (
  cmake -S '/github/home/ROOT-CI/src' -B '/github/home/ROOT-CI/build' -N -L
  )
  (
  cmake --build '/github/home/ROOT-CI/build' --config 'RelWithDebInfo' --parallel '16'
  )
  (
  cd '/github/home/ROOT-CI/build'
          ctest --output-on-failure --parallel 16 --output-junit TestResults.xml
  )
```

### ROOT version

master

### Installation method

docker copy/pasted from CI

### Operating system

Alma9 + Ubuntu

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dpiparo**
Fixes #16242

### CODE REVIEW

File: .github/workflows/root-ci.yml [hunk 1]
```diff
@@ -468,6 +468,7 @@ jobs:
         run: ".github/workflows/root-ci-config/build_root.py
                     --buildtype      ${{ inputs.buildtype }}
                     --platform       ${{ matrix.image }}
+                    --image          registry.cern.ch/root-ci/${{ matrix.image }}:buildready
```
Comment (pcanal) [inline .github/workflows/root-ci.yml:471]: Why is this related to improving the instructions?

File: .github/workflows/root-ci-config/build_utils.py [hunk 2]
```diff
@@ -76,6 +76,14 @@ def print(self) -> None:
                 print(f"""\
 Grab the image:
 $ docker run {' '.join(self.docker_opts)} -it {self.image}
+Setup the Python virtual environment after checking out root and roottest (see below)
+$ mkdir <ROOT build dir>
+$ cd <ROOT build dir>
```
Comment (pcanal) [inline .github/workflows/root-ci-config/build_utils.py:81]: Isn't this already done in the later instructions?

File: .github/workflows/root-ci-config/build_utils.py [hunk 3]
```diff
@@ -76,6 +76,14 @@ def print(self) -> None:
                 print(f"""\
 Grab the image:
 $ docker run {' '.join(self.docker_opts)} -it {self.image}
+Setup the Python virtual environment after checking out root and roottest (see below)
```
Comment (pcanal) [inline .github/workflows/root-ci-config/build_utils.py:79]: Rather than giving this instructions, couldn't those be embedded in the 'right' place below, so that copy/pasting the output would work?

Comment (dpiparo) [reply .github/workflows/root-ci-config/build_utils.py:79]: Not easily, since this is something done in the workflow and not in the script, the entity logging the commands. Therefore, we somehow need to remind to perform the steps necessary to set up the right python environment.

File: .github/workflows/root-ci-config/build_utils.py (hunk 2 above)
Comment (dpiparo) [reply .github/workflows/root-ci-config/build_utils.py:81]: correct, my bad.

File: .github/workflows/root-ci.yml (hunk 1 above)
Comment (dpiparo) [reply .github/workflows/root-ci.yml:471]: it can be inferred from the `--platform` option. Actually the full script can be simplified...

File: .github/workflows/root-ci-config/build_utils.py (hunk 3 above)
Comment (dpiparo) [reply .github/workflows/root-ci-config/build_utils.py]: also, there was a mistake. The dependencies are installed when building the container, and not before the build.

**Comment by dpiparo** [comment]
All comments above addressed. The failures are a glitch.

**Comment by dpiparo** [comment]
I tested ubuntu24 on lxplus and everything works out of the box with the new instructions.

File: .github/workflows/root-ci-config/build_utils.py [hunk 4]
```diff
@@ -74,9 +74,9 @@ def print(self) -> None:
         if self.trace != "":
             if self.image:
                 print(f"""\
-Grab the image:
-$ docker run {' '.join(self.docker_opts)} -it {self.image}
-Then:
+# Grab the image and set up the python virtual environment:
+docker run {' '.join(self.docker_opts)} -it registry.cern.ch/root-ci/{self.image}:buildready
+if [ -d /py-venv/ROOT-CI/bin/ ]; then . /py-venv/ROOT-CI/bin/activate && echo PATH=$PATH >> $GITHUB_ENV; fi
```
Comment (pcanal) [inline .github/workflows/root-ci-config/build_utils.py:79]: Suggestion:
```
# Grab the image 
docker run {' '.join(self.docker_opts)} -it registry.cern.ch/root-ci/{self.image}:buildready
# Then inside the container 
# Set up the python virtual environment:
if [ -d /py-venv/ROOT-CI/bin/ ]; then . /py-venv/ROOT-CI/bin/activate && echo PATH=$PATH >> $GITHUB_ENV; fi
